### PR TITLE
Clarify requirements

### LINKED ISSUE #106: collection_requirements are unclear with dependencies and version_added
##### SUMMARY
In overview/collection_requirements.rst
the lines:
"dependencies MUST be set to '>=1.0.0' (Might have to think about dependency order when publishing)"
and
"Use collection version numbers for version_added (otherwise version_added_collection must be provided, but this is discouraged)"
are rather cryptic. Please elaborate on their scope and meaning.

Is it
"versioned dependencies on other collections MUST be specified using the '>=' operator (e.g. namespace.collection>=1.0.0)"
and
"the 'version_added' field must be present and and refer to the corresponding collection version (version_added_collection SHOULD not be used anymore)"

##### ISSUE TYPE
- Bug Report

### PR DESCRIPTION
**Pull request by felixfontein**
##### SUMMARY
Fixes #106.

CC @bmillemathias  (can't request a review from you :) )

##### ISSUE TYPE
- Docs Pull Request

##### COMPONENT NAME
collection_requirements.rst

### CODE REVIEW

File: collection_requirements.rst [hunk 1]
```diff
@@ -41,7 +41,9 @@ galaxy.yml
 ----------
 
 * tags MUST be set
-* dependencies MUST be set to ``'>=1.0.0'`` (Might have to think about dependency order when publishing)
+* dependencies MUST be set to ``'>=1.0.0'``
+  * This means that all dependencies have to specify lower bounds on the versions, and these lower bounds should be proper releases, and not versions of the form 0.x.y.
```
Comment (bmillemathias) [inline collection_requirements.rst:45]: I don't understand "proper releases, and not versions of the form 0.x.y.".

File: collection_requirements.rst [hunk 2]
```diff
@@ -41,7 +41,9 @@ galaxy.yml
...
+  * When creating new collections where dependencies are also under development, you have to release the collection dependency first as 1.0.0, and then update the dependency version to 1.0.0 and publish the dependent collection as 1.0.0 since Galaxy checks whether the dependencies already exist on Galaxy.
```
Comment (bmillemathias) [inline collection_requirements.rst:46]: "dependencies" can be what kind ? collection or any type like python libraries?

File: collection_requirements.rst [hunk 3]
```diff
@@ -60,7 +62,9 @@ Documentation
 All module and plugin ``DOCUMENTATION`` and ``RETURN`` MUST:
 
 * Use the FQCN for ``M(...)`` and ``- module:`` references of ``seealso`` subsections. See `Linking within module documentation <https://docs.ansible.com/ansible/devel/dev_guide/developing_modules_documenting.html#linking-within-module-documentation>`_
-* Use collection version numbers for `version_added` (otherwise `version_added_collection` must be provided, but this is discouraged)
+* Use collection version numbers for ``version_added``, and not Ansible version numbers or other unrelated version numbers.
```
Comment (bmillemathias) [inline collection_requirements.rst:65]: should the sentence be rephrased
> use field `version_added` to define the version the module appeared in the collection. Don't define Ansible version numbers or other unrelated version numbers.

File: collection_requirements.rst [hunk 4]
```diff
@@ -60,7 +62,9 @@ Documentation
...
+  * If you for some reason really have to specify version numbers of Ansible or of another collection, you have to provide ``version_added_collection``. We strongly recommend to NOT do this.
+  * Not every option, module or plugin must have ``version_added``. You should use it to mark when new content (modules, plugins, options) were added to the module. The values are shown in the documentation, and this can be very useful for your users.
```
Comment (bmillemathias) [inline collection_requirements.rst:67]: > were added to the module.

shouldn'it be
> were added to the collection. 

?

Overall I don't understand the meaning between the 1st and the 2nd sentence which seem to me contradictory.

File: collection_requirements.rst (hunk 2 above)
Comment (felixfontein) [reply collection_requirements.rst:46]: galaxy.yml only has collection dependencies. I'll prefix it with `collection` though.

File: collection_requirements.rst (hunk 4 above)
Comment (felixfontein) [reply collection_requirements.rst:67]: Yes, thanks!

File: collection_requirements.rst (hunk 1 above)
Comment (felixfontein) [reply collection_requirements.rst:45]: I replaced `proper` by `stable`. Is it better this way?

File: collection_requirements.rst (hunk 3 above)
Comment (felixfontein) [reply collection_requirements.rst]: I adjusted the sentence. Can you take another look?

File: collection_requirements.rst (hunk 1 above)
Comment (bmillemathias) [reply collection_requirements.rst]: ah yes !

**Comment by felixfontein** [comment]
@bmillemathias @gundalow thanks for your feedback and merging!

**Pull request merged**